nemo/collections/asr/contextnet.py

```diff
@@ -0,0 +1,275 @@
+                        # or only the last sub-block will perform a strided convolution.
+                }
+
+        activation (str): Activation function used for each sub-blocks. Can be
+            one of ["hardtanh", "relu", "selu", "swish"].
+        feat_in (int): Number of channels being input to this module
+        normalization_mode (str): Normalization to be used in each sub-block.
+            Can be one of ["batch", "layer", "instance", "group"]
+            Defaults to "batch".
+        residual_mode (str): Type of residual connection.
+            Can be "add", "stride_add" or "max".
+            "stride_add" mode performs strided convolution prior to residual
+            addition.
+            Defaults to "add".
+        norm_groups (int): Number of groups for "group" normalization type.
+            If set to -1, number of channels is used.
+            Defaults to -1.
+        conv_mask (bool): Controls the use of sequence length masking prior
+            to convolutions.
+            Defaults to True.
+        frame_splicing (int): Defaults to 1.
+        init_mode (str): Describes how neural network parameters are
+            initialized. Options are ['xavier_uniform', 'xavier_normal',
+            'kaiming_uniform','kaiming_normal'].
+            Defaults to "xavier_uniform".
+    """
+
+    length: Optional[torch.Tensor]
+
+    @property
+    @add_port_docs()
+    def input_ports(self):
+        """Returns definitions of module input ports.
+        """
+        return {
+            # "audio_signal": NeuralType(
+            #    {0: AxisType(BatchTag), 1: AxisType(SpectrogramSignalTag), 2: AxisType(ProcessedTimeTag),}
+            # ),
+            # "length": NeuralType({0: AxisType(BatchTag)}),
+            "audio_signal": NeuralType(('B', 'D', 'T'), SpectrogramType()),
+            "length": NeuralType(tuple('B'), LengthsType()),
+        }
+
+    @property
+    @add_port_docs()
+    def output_ports(self):
+        """Returns definitions of module output ports.
+        """
+        return {
+            # "outputs": NeuralType(
+            #    {0: AxisType(BatchTag), 1: AxisType(EncodedRepresentationTag), 2: AxisType(ProcessedTimeTag),}
+            # ),
+            # "encoded_lengths": NeuralType({0: AxisType(BatchTag)}),
+            "outputs": NeuralType(('B', 'D', 'T'), AcousticEncodedRepresentation()),
+            "encoded_lengths": NeuralType(tuple('B'), LengthsType()),
+        }
+
+    @property
+    def _disabled_deployment_input_ports(self):
+        return set(["length"])
+
+    @property
+    def _disabled_deployment_output_ports(self):
+        return set(["encoded_lengths"])
+
+    def _prepare_for_deployment(self):
+        m_count = 0
+        for m in self.modules():
+            if type(m).__name__ == "MaskedConv1d":
+                m.use_mask = False
+                m_count += 1
+        logging.warning(f"Turned off {m_count} masked convolutions")
+
+    def __init__(
+        self,
```

Comment:
for new Modules/Classes please let's start using Python 3 type hints

Reply:
Sounds good.

Reply:
Fixed